ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -80,15 +84,15 @@ public AlterTableCompactOperation(DDLOperationContext context, AlterTableCompact
       Optional<String> partitionName =  partitionMap.keySet().stream().findFirst();
       partitionName.ifPresent(compactionRequest::setPartitionname);
     }
-    List<CompactionResponse> compactionResponses =
+    Map<CompactionResponse, String> compactionResponses =
         initiatorBase.initiateCompactionForTable(compactionRequest, table.getTTable(), partitionMap);
-    for (CompactionResponse compactionResponse : compactionResponses) {
+    for (Map.Entry<CompactionResponse, String> compactionResponseEntry : compactionResponses.entrySet()) {
```

Comment:
Instead of creating and passing a `Map` around why don't we extract the partition iteration here instead of having it inside the `initiateCompactionForTable`? This way we will not need a Map and will avoid the need for iterating over the partitions twice (once inside the `initiateCompactionForTable` and once over the respective responses here).